gprofiler/metadata/application_identifiers.py

```diff
@@ -0,0 +1,256 @@
+    """
+    args = process.cmdline()
+
+    # Just to be on the safe side, module name must be run after the "-m" argument, python enforces it.
+    if len(args) < 3:
+        return False
+
+    return _is_python_bin(args[0]) and args[1] == "-m"
+
+
+def _is_python_bin(bin_name: str) -> bool:
+    return _PYTHON_BIN_RE.match(os.path.basename(bin_name)) is not None
+
+
+def _get_cli_arg_by_name(args: List[str], arg_name: str, check_for_equals_arg: bool = False) -> str:
+    if arg_name in args:
+        return args[args.index(arg_name) + 1]
+
+    if check_for_equals_arg:
+        for arg in args:
+            arg_key, _, arg_val = arg.rpartition("=")
+            if arg_key == arg_name:
+                return arg_val
+
+    return _NON_AVAILABLE_ARG
+
+
+def _get_cli_arg_by_index(args: List[str], index: int) -> str:
+    try:
+        return args[index]
+    except IndexError:
+        return _NON_AVAILABLE_ARG
+
+
+def _append_python_module_to_proc_wd(process: Process, module: str) -> str:
+    # Convert module name to module path, for example a.b -> a/b.py
+    if not module.endswith(".py"):
+        module = module.replace(".", "/") + ".py"
+
+    if os.path.isabs(module):
+        return module
+
+    return os.path.realpath(os.path.join(process.cwd(), module))
+
+
+class _ApplicationIdentifier(metaclass=ABCMeta):
+    @abstractmethod
+    def get_application_name(self, process: Process) -> Optional[str]:
+        pass
+
+
+class _GunicornApplicationIdentifier(_ApplicationIdentifier):
+    def get_application_name(self, process: Process) -> Optional[str]:
+        # As of gunicorn documentation the WSGI module name most probably will come from the cmdline and not from the
+        # config file / environment variables (they added the option to specify `wsgi_app`
+        # in the config file only in version 20.1.0)
+
+        if "gunicorn" != os.path.basename(
+            _get_cli_arg_by_index(process.cmdline(), 0)
+        ) and "gunicorn" != os.path.basename(_get_cli_arg_by_index(process.cmdline(), 1)):
+            return None
+
+        # wsgi app specification will come always as the last argument (if hasn't been specified config file)
+        wsgi_app_spec = process.cmdline()[-1].split(":", maxsplit=1)[0]
+        return f"gunicorn: {_append_python_module_to_proc_wd(process, wsgi_app_spec)}"
+
+
+class _UwsgiApplicationIdentifier(_ApplicationIdentifier):
+    @staticmethod
+    def _find_wsgi_from_config_file(process: Process) -> Optional[str]:
+        for arg in process.cmdline():
+            if arg.endswith(".ini"):
+                config_file = arg
+                break
+        else:
+            return None
+
+        if not os.path.isabs(config_file):
+            config_file = os.path.join(process.cwd(), config_file)
+
+        config = configparser.ConfigParser()
+        config.read(resolve_host_path(process, config_file))
+        try:
+            return config.get("uwsgi", "module")
```

Comment:
What do the parameters mean here? It looks like a dict access with a default value, but you're catching some exceptions, so?

Reply:
Yep, it's the `ConfigParser` (builtin parser for ini files), added a comment

Reply:
kk